sdk/helper/tokenutil/root/encode.go

```diff
@@ -0,0 +1,48 @@
+package root
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/helper/pgpkeys"
+	"github.com/hashicorp/vault/sdk/helper/xor"
+)
+
+var (
+	// ErrNoTokenProvided is returned when an empty string is provided as an error
+	ErrNoTokenProvided = errors.New("no token provided")
```

Comment:
We don't usually define error types unless we have to, i.e. because some code elsewhere needs to match the error and do something specific when that error is returned.

Reply:
I defined here the errors so that I can compare them more accurately in the tests, but I don't have a strong opinion on this. Will change